tower-http/src/compression/future.rs

```diff
@@ -21,14 +25,43 @@ pub struct ResponseFuture<F> {
 
 impl<F, B, E> Future for ResponseFuture<F>
 where
-    F: Future<Output = Result<http::Response<B>, E>>,
+    F: Future<Output = Result<Response<B>, E>>,
     B: Body,
+    B::Error: Into<Box<dyn std::error::Error + Send + Sync>>,
```

Comment:
might be worth it to just define a `type BoxError`

Reply:
36f8084acff29 👍